src/test/rgw/rgw_multi/tests.py

```diff
@@ -459,24 +472,36 @@ def check_bucket_eq(zone1, zone2, bucket_name):
         log.debug('o=%s', o.name)
 
     for k1, k2 in zip_longest(b1.get_all_versions(), b2.get_all_versions()):
+        if k1 is None and k2 is None:
+            return
         if k1 is None:
             log.critical('key=%s is missing from zone=%s', k2.name, zone1.name)
             assert False
         if k2 is None:
             log.critical('key=%s is missing from zone=%s', k1.name, zone2.name)
             assert False
 
-        check_object_eq(k1, k2)
+        if isinstance(k1, DeleteMarker):
+            dm.add(k1.name)
+            if isinstance(k2, DeleteMarker):
+                check_deletemarker_eq(k1, k2)
+            else:
+                assert False
+        else:
+            if isinstance(k2, DeleteMarker):
+                assert Fasle
+            if k1.name in dm:
+                continue
```

Comment:
can you explain what `dm` is for here? why avoid `check_object_eq()` if there's a delete marker with that name?